core/services/workflows/syncer/handler.go

```diff
@@ -177,11 +179,16 @@ func NewEventHandler(
 		o(eh)
 	}
 
-	return eh
+	if engineRegistry == nil {
```

Comment:
Nit: this is a bit strange -- you've set up the struct incorrectly (with a nil registry) and then proceed to validate; future extensions could end up making changes that break this line

Much better to move this to the very top as the first thing you check

Reply:
I originally had it like that, but @MStreet3  pointed out that the options functions may introduce changes to the struct. Thus moving it down.